gnome-initial-setup/pages/summary/gis-summary-page.c

```diff
@@ -236,6 +239,117 @@ done_cb (GtkButton *button, GisSummaryPage *page)
     }
 }
 
+static void
+initial_contact_app_signal (GDBusProxy *proxy,
+                            gchar *sender,
+                            gchar *signal_name,
+                            GVariant *parameters,
+                            gpointer user_data)
+{
+  GisSummaryPage *page = user_data;
+  GisSummaryPagePrivate *priv = gis_summary_page_get_instance_private (page);
+  const gchar *object, *property;
+  g_autoptr(GVariant) value;
+
+  if (g_strcmp0 (signal_name, "ObjectNotify") != 0)
+    return;
+
+  g_variant_get (parameters, "(&s&sv)", &object, &property, &value);
+
+  if (g_strcmp0 (object, "view.JSContext.globalParameters") != 0)
+    return;
+
+  if (g_strcmp0 (property, "unlocked") != 0)
+    return;
+
+  g_return_if_fail (g_variant_is_of_type (value, G_VARIANT_TYPE_BOOLEAN));
+
+  if (g_variant_get_boolean (value))
+    {
+      g_signal_handler_disconnect (priv->clippy_proxy,
+                                   priv->clippy_proxy_signal_id);
+      finish_fbe (page);
+    }
+}
+
+static void
+initial_contact_connect_to_clippy (GisSummaryPage *page)
+{
+  GisSummaryPagePrivate *priv = gis_summary_page_get_instance_private (page);
+  g_autoptr(GError) error = NULL;
+
+  g_dbus_proxy_call_sync (priv->clippy_proxy,
+                          "Connect",
```

Comment:
Do we need to call this? I thought we'd get the `ObjectNotify` signal in any case when it's changed from the app.